src/argilla/training/setfit.py

```diff
@@ -0,0 +1,157 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import json
+from typing import List, Union
+
+from datasets import DatasetDict
+
+from argilla.training.utils import filter_allowed_args, get_default_args
+
+
+class ArgillaSetFitTrainer(object):
+    # @require_version("setfit", "0.6")
+    def __init__(self, dataset, device, record_class, multi_label: bool = False, seed: int = None):
+        from setfit import SetFitModel, SetFitTrainer
+
+        if seed is None:
+            seed = 42
+        self._seed = seed
+        self._record_class = record_class
+        self._multi_label = multi_label
+
+        if isinstance(dataset, DatasetDict):
+            self._train_dataset = dataset["train"]
+            self._eval_dataset = dataset["test"]
```

Comment:
This might require some documentation somewhere

Reply:
How so?

Reply:
Not all datasets use `train` and `test` (exclusively). E.g. `dev`, `eval`, `validation`, etc.

Reply:
That is true but the dataset is inherited from the `prepare_for_training` argilla method, which only supports "train", "test",  but I will add some comments to clarify.

Reply:
Oh, then it should be fine I reckon.